src/StructuredLogger/ObjectModel/TreeNode.cs

```diff
@@ -47,6 +47,8 @@ public bool IsExpanded
             }
         }
 
+        private Dictionary<ChildrenCacheKey, object> childrenCache;

```

Comment:
GetOrCreateNodeWithName needs access to cache, so you can not just store IList and access cache at the same time.